apps/emqx_mysql/src/emqx_mysql.erl

```diff
@@ -267,15 +272,17 @@ do_check_prepares(
     );
 do_check_prepares(#{prepares := ok}) ->
     ok;
-do_check_prepares(#{prepares := {error, _}} = State) ->
+do_check_prepares(#{prepares := {error, _}, query_templates := _} = State) ->
```

Comment:
Q: is there a reason to assert this key is present?  It doesn't seem to be used in other clauses of this function.

Reply:
The reason is to skip to the (new) last clause, in case it is missing. That wasn't necessary in the old implementation but now, if there are no channels added (yet) then there's also no `query_templates`.